auth/jaspi/src/main/java/org/wildfly/security/auth/jaspi/DelegatingAuthConfigFactory.java

```diff
@@ -53,7 +53,7 @@ public DelegatingAuthConfigFactory(final ElytronAuthConfigFactory elytronAuthCon
     @Override
     public AuthConfigProvider getConfigProvider(String layer, String appContext, RegistrationListener listener) {
         AuthConfigProvider authConfigProvider = elytronAuthConfigFactory.getConfigProvider(layer, appContext, listener);
-        if (authConfigProvider != null || elytronAuthConfigFactory.matchesRegistration(layer, appContext) || !delegationAllowed.get()) {
+        if (authConfigProvider != null || elytronAuthConfigFactory.matchesRegistration(layer, appContext) || Boolean.TRUE.equals(delegationAllowed.get())) {
```

Comment:
Because the original condition was `!delegationAllowed.get()`, this should be updated to either:

* !Boolean.TRUE.equals(delegationAllowed.get())

or more directly:

* `Boolean.FALSE.equals(delegationAllowed.get())`.